pkg/cortex/cortex.go

```diff
@@ -185,38 +185,39 @@ func (c *Config) Validate(log log.Logger) error {
 
 // Cortex is the root datastructure for Cortex.
 type Cortex struct {
-	target ModuleName
+	Cfg Config
 
 	// set during initialization
-	serviceMap map[ModuleName]services.Service
-
-	api              *api.API
-	server           *server.Server
-	ring             *ring.Ring
-	overrides        *validation.Overrides
-	distributor      *distributor.Distributor
-	ingester         *ingester.Ingester
-	flusher          *flusher.Flusher
-	store            chunk.Store
-	deletesStore     *purger.DeleteStore
-	frontend         *frontend.Frontend
-	tableManager     *chunk.TableManager
-	cache            cache.Cache
-	runtimeConfig    *runtimeconfig.Manager
-	dataPurger       *purger.DataPurger
-	tombstonesLoader *purger.TombstonesLoader
-
-	ruler        *ruler.Ruler
-	configAPI    *configAPI.API
-	configDB     db.DB
-	alertmanager *alertmanager.MultitenantAlertmanager
-	compactor    *compactor.Compactor
-	storeGateway *storegateway.StoreGateway
-	memberlistKV *memberlist.KVInit
+	ServiceMap    map[string]services.Service
+	ModuleManager *modules.Manager
+
+	API              *api.API
+	Server           *server.Server
+	Ring             *ring.Ring
+	Overrides        *validation.Overrides
+	Distributor      *distributor.Distributor
+	Ingester         *ingester.Ingester
+	Flusher          *flusher.Flusher
+	Store            chunk.Store
+	DeletesStore     *purger.DeleteStore
+	Frontend         *frontend.Frontend
+	TableManager     *chunk.TableManager
+	Cache            cache.Cache
+	RuntimeConfig    *runtimeconfig.Manager
+	DataPurger       *purger.DataPurger
+	TombstonesLoader *purger.TombstonesLoader
```

Comment:
Are renames important for this PR? Let's keep them as a separate PR to minimize amount of changes in this one.

Reply:
@annanay25 Can you open a separate PR that just handles renaming. Since there is no functional change it should be quick to review and lower the surface area of this PR.